### ISSUE TITLE
docker_network cdef config error

### ISSUE THREAD

**Issue by jrmsgit** [munin-monitoring/contrib#1182]
Hi:

I'm getting the following errors on munin-graph.log trying to use docker_ as docker_network plugin.

2021/02/22 12:45:21 [RRD ERROR] Unable to graph /var/cache/munin/www/local/testing.local/docker_network-week.png : don't understand 'testing_down,8,*'

    # munin-run docker_network config
    ...
    ...
    testing_down.cdef testing_down,8,*
    ...
    ...

I tried some different settings for .cdef but I couldn't make it work so I can't provide a patch for the fix sorry. I'm not sure how that should be configured.

Let me know please if you need any extra information or anything I could help you with.

Thanks all for the great job on this project.

Cheers,

Jeremías

**Comment by sumpfralle** [comment]
```
2021/02/22 12:45:21 [RRD ERROR] Unable to graph /var/cache/munin/www/local/testing.local/docker_network-week.png : don't understand 'testing_down,8,*'
```

This is quite an unexpected error message, since rrdtool should easily be able to parse this value.
Indeed I configured the plugin on my local host and was able to view the resulting graphs.

Maybe you are using a *really* ancient version of rrd?
Or do you have other ideas, what could cause problems?

**Comment by jrmsgit** [comment]
Hi:

Thanks for your reply!

> Maybe you are using a really ancient version of rrd?

I'm using munin-node and deps from debian testing official repo.

munin-node 2.0.66-1
rrdtool 1.7.2-3+b7

Not sure how old that rrdtool version is.

> Or do you have other ideas, what could cause problems?

Sorry, my bad, I changed the containers name to testing for the issue report not to use the production name, just in case. But it turns out that I just checked using testing, testing_one, and testing-two as container names (run --name) and the only one failing is testing-two. The production name I'm using has dashes in its name too (like: aaa-bbb-ccc), so I think that's the issue here.

I know now and I can change the containers name, but it would be good if the plugin can deal with that?

Maybe use a sanitized version of container's name replacing '-' symbols and maybe others too for plugin's config instead of directly using its name? Maybe gen a unique id from container's name? Not sure what could be used which generates a valid rrd cdef statement for the variable name.

Let me know please if I could be of any help.

Thanks,


Jeremías

**Comment by sumpfralle** [comment]
I fixed this now in be4ad90df32204713f51c3e5cf7b8fc7816d3ba0.

Thanks for your report!

**Comment by jrmsgit** [comment]
Thanks you for the quick fix!!
